### PR TITLE
Cybereason xsoar v 2.1.14

### PR DESCRIPTION
**Pull request by suraj-metron**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by sapirshuker** [comment]
Hey, @suraj-metron is the PR ready for review?

**Comment by suraj-metron** [comment]
Hi @sapirshuker ,
Yes, the PR is ready for review.
It contains all the comments fixed from older comments.
Thanks!

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py [hunk 1]
```diff
@@ -1910,6 +1912,35 @@ def get_sensor_id_command(client: Client, args: dict):
...
+    else:
+        outputs = []
+        for single_sensor in response.get('sensors'):
+            if single_sensor.get('machineName') == machine_name:
+                outputs.append({
+                    "MachineID": single_sensor["sensorId"],
+                    "MachineName": single_sensor["machineName"],
+                    "MachineFQDN": single_sensor["fqdn"],
+                    "GroupID": single_sensor["groupId"],
+                    "GroupName": single_sensor["groupName"]
```
Comment (sapirshuker) [inline Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1933]:
```suggestion
                outputs.append({
                    "MachineID": single_sensor.get("sensorId"),
                    "MachineName": single_sensor.get("machineName"),
                    "MachineFQDN": single_sensor.get("fqdn"),
                    "GroupID": single_sensor.get("groupId"),
                    "GroupName": single_sensor.get("groupName")
```

**Review by sapirshuker — CHANGES_REQUESTED** [review]
Hi @suraj-metron the code looks good!
Just a small change and we're ready for a demo. 
Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR or in email.

**Comment by suraj-metron** [comment]
Hi @sapirshuker 
Thank you for the suggested code change. I have made the relevant code change.
For the Demo, I will connect to you via email.

Thanks!

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py [hunk 2]
```diff
@@ -1910,6 +1912,35 @@ def get_sensor_id_command(client: Client, args: dict):
         return CommandResults(readable_output=f"Available Sensor IDs are {output}")
 
 
+def fetch_machine_details_command(client: Client, args: dict):
+    machine_name = str(args.get('machineName'))
+    json_body = {
```
Comment (sapirshuker) [inline Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1917]: Hey, what do you think about adding an argument of limit? 
what happened if the user have more then 1000 results?

**Review by sapirshuker — CHANGES_REQUESTED** [review]

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py (hunk 2 above)
Comment (sapirshuker) [reply Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1917]: My apologies for missing it.

Comment (suraj-metron) [reply Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1917]: Hi @sapirshuker 
Sure, we shall include the limit argument as in very extreme cases there might be a possibility of the user having 1000 machines with same name

Comment (sapirshuker) [reply Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1917]: Hey, @suraj-metron 
attached the best practice for [pagination in integration commands](  https://xsoar.pan.dev/docs/integrations/code-conventions#pagination-in-integration-commands )

Comment (suraj-metron) [reply Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1917]: Hi @sapirshuker 
Thank you for the guide. I shall follow it and will make the necessary changes

**Comment by sapirshuker** [comment]
Hi @suraj-metron , we haven’t heard from you in a while.
Do you need any help with the pull request?

Please feel free to reach out to me here or on Slack.
Thanks again for contributing to our repo, hope to hear from you soon.

**Comment by suraj-metron** [comment]
> Hi @suraj-metron , we haven’t heard from you in a while. Do you need any help with the pull request?
> 
> Please feel free to reach out to me here or on Slack. Thanks again for contributing to our repo, hope to hear from you soon.

Hi @sapirshuker 
We are working on it. The team was on a public holiday. 
Thank you for your help, much appreciated.
Once we are done with the changes, I shall update you.

Thanks!

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py [hunk 3]
```diff
@@ -2041,6 +2103,9 @@ def main():
         elif demisto.command() == 'cybereason-get-sensor-id':
             return_results(get_sensor_id_command(client, args))
 
+        elif demisto.command() == 'cybereason-get-machine-details':
+            return_results(fetch_machine_details_command(client, args))
```
Comment (sapirshuker) [inline Packs/Cybereason/Integrations/Cybereason/Cybereason.py:2107]: Please change the function name to align with the command name.

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py [hunk 4]
```diff
@@ -1910,6 +1912,66 @@ def get_sensor_id_command(client: Client, args: dict):
...
+        empty_output_message = 'No Machine Details found for the given Machine Name'
+        return CommandResults(
+            readable_output=tableToMarkdown(
+                'Machine Details', outputs, headers=SENSOR_HEADERS) if outputs else empty_output_message,
+            outputs_prefix='Cybereason.Sensor',
+            outputs_key_field='MachineID',
+            outputs=outputs)
+
+
+def fetch_machine_details_command_pagination_params(args: dict) -> dict:
```
Comment (sapirshuker) [inline Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1941]: Please change the function name to align with the command name.

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py [hunk 5]
```diff
@@ -1910,6 +1912,66 @@ def get_sensor_id_command(client: Client, args: dict):
         return CommandResults(readable_output=f"Available Sensor IDs are {output}")
 
 
+def fetch_machine_details_command(client: Client, args: dict):
+    machine_name = str(args.get('machineName'))
+    json_body = fetch_machine_details_command_pagination_params(args)
```
Comment (sapirshuker) [inline Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1917]: You didn't add the machine_name to the request.

File: Packs/Cybereason/Playbooks/playbook-Unisolate_Endpoint_-_Cybereason.yml [hunk 6]
```diff
@@ -25,7 +25,6 @@ tasks:
     ignoreworker: false
     skipunavailable: false
```
Comment (sapirshuker) [inline Packs/Cybereason/Playbooks/playbook-Unisolate_Endpoint_-_Cybereason.yml:26]: Did you include this change? Do we need it?
If it's not necessary, please revert it.

**Review by sapirshuker — CHANGES_REQUESTED** [review]
Hi @suraj-metron
Thank you for your contribution!
Great work !! 
Please see my comments.

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py (hunk 3 above)
Comment (suraj-metron) [reply Packs/Cybereason/Integrations/Cybereason/Cybereason.py:2107]: Done

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py (hunk 4 above)
Comment (suraj-metron) [reply Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1941]: Done

File: Packs/Cybereason/Integrations/Cybereason/Cybereason.py (hunk 5 above)
Comment (suraj-metron) [reply Packs/Cybereason/Integrations/Cybereason/Cybereason.py:1917]: We are not adding it to the API request, we are filtering the data after we have received the response

File: Packs/Cybereason/Playbooks/playbook-Unisolate_Endpoint_-_Cybereason.yml (hunk 6 above)
Comment (suraj-metron) [reply Packs/Cybereason/Playbooks/playbook-Unisolate_Endpoint_-_Cybereason.yml:26]: Hi, There were some demisto validation errors for this Playbook. So, just fixed those errors.

**Review by sapirshuker — COMMENTED** [review]
Hi @suraj-metron , the code looks good!
We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep). The demo was scheduled for 28/11 17:30 GMT +2

**Comment by suraj-metron** [comment]
Hi @sapirshuker 
Thank you for scheduling the demo. We shall go through the demo guidelines and be available for the demo
Thanks!

**Comment by sapirshuker** [comment]
XSIAM pipeline failed on known issue.

**Review by sapirshuker — APPROVED** [review]
Great Job!!

**Pull request merged**